### DISCUSSION TITLE
Problem preview mode + `server$` in 0.100.0 qwik

### DISCUSSION THREAD

**Discussion by ahnpnl** [robisim74/qwik-speak#41] [Q&A]
When I tried **0.10.0** and put `loadTranslation$` into a separate file with the content as below
```
const translationData = import.meta.glob('/assets/i18n/**/*.json');

const loadTranslation$: LoadTranslationFn = server$((lang: string, asset: string) =>
    translationData[`/assets/i18n/${lang}/${asset}.json`]?.(),
);

export const translationFn: TranslationFn = {
    loadTranslation$,
};
```

When I ran preview mode, the app didn't load anymore and I ran into exception
```
TypeError: Invalid value used in weak set
    at WeakSet.add (<anonymous>)
    at noSerialize (file:///Users/anh.pham/branded-pages/node_modules/@builder.io/qwik/core.mjs:7722:24)
    at loadTranslations (file:///Users/anh.pham/branded-pages/node_modules/qwik-speak/lib/index.qwik.mjs:38:40)
```
seem like there is an issue with `noSerialize` function with **0.100.0** qwik

Investigated more, I found out that the error will throw out when `noSerializer` executes with a normal object as input. Rollback to **0.25.0** qwik also saw the same thing.

**Comment by robisim74** [comment]
Could you post an example of your json file?

Are you are using json with depth 0?

**Reply by ahnpnl** [reply]
The json is 
```
{
    "foo": "bar"
}
```
or without the json file, if we make the `loadTranslation$` function like this
```
const loadTranslation$: LoadTranslationFn = server$((lang, asset) => {
     return {
            "foo": "bar"
     }
});
```
the error also occurs

Indeed JSON with depth 0 causes the issue. I wonder why `noSerialize` doesn't allow depth 0, weird

**Reply by robisim74** [reply]
Yes, make no sense depth 0 for serialization. In all examples of this library I suggest to use scoped translations, where the level 0 is the name of the asset. But ok, I should cover also the case of depth 0. This is a bug.

**Comment by robisim74** [comment]
Should be fixed in v.0.10.1
